dnsdist 2.0.0-rc1 fails to build on 32bit

<!-- Hi! Thanks for filing an issue. It will be read with care by human beings. Can we ask you to please fill out this template and not simply demand new features or send in complaints? Thanks! -->
<!-- Also please search the existing issues (both open and closed) to see if your report might be duplicate -->
- [x] This is not a support question, I have read [about opensource](https://www.powerdns.com/opensource.html) and will send support questions to the IRC channel, [GitHub Discussions](https://github.com/PowerDNS/pdns/discussions/) or the mailing list.
- [x] I have read and understood the ['out in the open' support policy](https://blog.powerdns.com/2016/01/18/open-source-support-out-in-the-open/)

<!-- Tell us what is issue is about -->
 - Program: dnsdist <!-- delete the ones that do not apply -->
 - Issue type: Bug report

### Short description
<!-- Explain in a few sentences what the issue/request is -->

```
dnsdist-lua-configuration-items.cc:41:65: error: narrowing conversion of ‘18446744073709551615’ from ‘long long unsigned int’ to ‘size_t’ {aka ‘unsigned int’} [-Wnarrowing]
   41 |   const size_t maximumValue{std::numeric_limits<uint64_t>::max()};
      |                                                                 ^
dnsdist-lua-configuration-items.cc:56:65: error: narrowing conversion of ‘18446744073709551615’ from ‘long long unsigned int’ to ‘size_t’ {aka ‘unsigned int’} [-Wnarrowing]
   56 |   const size_t maximumValue{std::numeric_limits<uint64_t>::max()};
      |                                                                 ^
dnsdist-lua-configuration-items.cc:92:195: error: narrowing conversion of ‘18446744073709551615’ from ‘long long unsigned int’ to ‘size_t’ {aka ‘unsigned int’} [-Wnarrowing]
   92 |   {"setMaxTCPQueriesPerConnection", {[](dnsdist::configuration::RuntimeConfiguration& config, uint64_t newValue) { config.d_maxTCPQueriesPerConn = newValue; }, std::numeric_limits<uint64_t>::max()}},
      |                                                                                                                                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
dnsdist-lua-configuration-items.cc:133:203: error: narrowing conversion of ‘18446744073709551615’ from ‘long long unsigned int’ to ‘size_t’ {aka ‘unsigned int’} [-Wnarrowing]
  133 |   {"setMaxTCPConnectionsPerClient", {[](dnsdist::configuration::ImmutableConfiguration& config, uint64_t newValue) { config.d_maxTCPConnectionsPerClient = newValue; }, std::numeric_limits<uint64_t>::max()}},
      |                                                                                                                                                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
make[4]: *** [Makefile:1931: dnsdist-lua-configuration-items.o] Error 1
make[4]: *** Waiting for unfinished jobs....
```

full build log:
https://buildd.debian.org/status/fetch.php?pkg=dnsdist&arch=armel&ver=2.0.0~rc1-2&stamp=1752356297&file=log

Can be seen on armel, armhf, i386


dnsdist 1.9.10 built fine on these archs.


### Environment
<!-- Tell us about the environment -->
 - Operating system: Debian unstable
 - Software version: 2.0.0-rc1
 - Software source: source <!-- e.g. Operating system repository, PowerDNS repository, compiled yourself -->



### Other information
<!-- if you already did more digging into the issue, please provide all the information you gathered -->

Toolchain package versions: binutils_2.44-3 dpkg-dev_1.22.21 g++-14_14.2.0-19 gcc-14_14.2.0-19 libc6-dev_2.41-9 libstdc++-14-dev_14.2.0-19 libstdc++6_14.2.0-19 linux-libc-dev_6.12.35-1


Package versions: apt_3.0.3 apt-cudf_7.0.0-6+b1 aspcud_1:1.9.6-2 autoconf_2.72-3.1 automake_1:1.17-4 autopoint_0.23.1-2 autotools-dev_20240727.1 base-files_13.8 base-passwd_3.6.7 bash_5.2.37-2+b3 binutils_2.44-3 binutils-arm-linux-gnueabi_2.44-3 binutils-common_2.44-3 bsdextrautils_2.41-5 bsdutils_1:2.41-5 build-essential_12.12 bzip2_1.0.8-6 ca-certificates_20250419 cargo_1.85.0+dfsg3-1 clasp_3.3.5-4.2 coreutils_9.7-3 cpp_4:14.2.0-1 cpp-14_14.2.0-19 cpp-14-arm-linux-gnueabi_14.2.0-19 cpp-arm-linux-gnueabi_4:14.2.0-1 cxxbridge-cmd_1.0.141-1+b1 dash_0.5.12-12 debconf_1.5.91 debhelper_13.24.2 debian-archive-keyring_2025.1 debianutils_5.23.1 dh-autoreconf_20 dh-cargo_32 dh-strip-nondeterminism_1.14.1-2 diffutils_1:3.10-4 dpkg_1.22.21 dpkg-dev_1.22.21 dwz_0.15-1+b3 fakeroot_1.37.1.2-1 file_1:5.46-5 findutils_4.10.0-3 g++_4:14.2.0-1 g++-14_14.2.0-19 g++-14-arm-linux-gnueabi_14.2.0-19 g++-arm-linux-gnueabi_4:14.2.0-1 gcc_4:14.2.0-1 gcc-14_14.2.0-19 gcc-14-arm-linux-gnueabi_14.2.0-19 gcc-14-base_14.2.0-19 gcc-arm-linux-gnueabi_4:14.2.0-1 gettext_0.23.1-2 gettext-base_0.23.1-2 gfortran-14_14.2.0-19 gfortran-14-arm-linux-gnueabi_14.2.0-19 grep_3.11-4 gringo_5.6.2-2 groff-base_1.23.0-9 gzip_1.13-1 hostname_3.25 hwloc-nox_2.12.0-4 icu-devtools_76.1-4 init-system-helpers_1.68 intltool-debian_0.35.0+20060710.6 libabsl-dev_20240722.0-4 libabsl20240722_20240722.0-4 libacl1_2.3.2-2+b1 libapparmor1_4.1.0-1 libapt-pkg7.0_3.0.3 libarchive-zip-perl_1.68-1 libasan8_14.2.0-19 libatomic1_14.2.0-19 libattr1_1:2.5.2-3 libaudit-common_1:4.0.5-1 libaudit1_1:4.0.5-1 libbinutils_2.44-3 libblkid1_2.41-5 libboost-all-dev_1.83.0.2+b2 libboost-atomic-dev_1.83.0.2+b2 libboost-atomic1.83-dev_1.83.0-4.2 libboost-atomic1.83.0_1.83.0-4.2 libboost-chrono-dev_1.83.0.2+b2 libboost-chrono1.83-dev_1.83.0-4.2 libboost-chrono1.83.0t64_1.83.0-4.2 libboost-container-dev_1.83.0.2+b2 libboost-container1.83-dev_1.83.0-4.2 libboost-container1.83.0_1.83.0-4.2 libboost-context-dev_1.83.0.2+b2 libboost-context1.83-dev_1.83.0-4.2 libboost-context1.83.0_1.83.0-4.2 libboost-coroutine-dev_1.83.0.2+b2 libboost-coroutine1.83-dev_1.83.0-4.2 libboost-coroutine1.83.0_1.83.0-4.2 libboost-date-time-dev_1.83.0.2+b2 libboost-date-time1.83-dev_1.83.0-4.2 libboost-date-time1.83.0_1.83.0-4.2 libboost-dev_1.83.0.2+b2 libboost-exception-dev_1.83.0.2+b2 libboost-exception1.83-dev_1.83.0-4.2 libboost-fiber-dev_1.83.0.2+b2 libboost-fiber1.83-dev_1.83.0-4.2 libboost-fiber1.83.0_1.83.0-4.2 libboost-filesystem-dev_1.83.0.2+b2 libboost-filesystem1.83-dev_1.83.0-4.2 libboost-filesystem1.83.0_1.83.0-4.2 libboost-graph-dev_1.83.0.2+b2 libboost-graph-parallel-dev_1.83.0.2+b2 libboost-graph-parallel1.83-dev_1.83.0-4.2 libboost-graph-parallel1.83.0_1.83.0-4.2 libboost-graph1.83-dev_1.83.0-4.2 libboost-graph1.83.0_1.83.0-4.2 libboost-iostreams-dev_1.83.0.2+b2 libboost-iostreams1.83-dev_1.83.0-4.2 libboost-iostreams1.83.0_1.83.0-4.2 libboost-json-dev_1.83.0.2+b2 libboost-json1.83-dev_1.83.0-4.2 libboost-json1.83.0_1.83.0-4.2 libboost-locale-dev_1.83.0.2+b2 libboost-locale1.83-dev_1.83.0-4.2 libboost-locale1.83.0_1.83.0-4.2 libboost-log-dev_1.83.0.2+b2 libboost-log1.83-dev_1.83.0-4.2 libboost-log1.83.0_1.83.0-4.2 libboost-math-dev_1.83.0.2+b2 libboost-math1.83-dev_1.83.0-4.2 libboost-math1.83.0_1.83.0-4.2 libboost-mpi-dev_1.83.0.2+b2 libboost-mpi-python-dev_1.83.0.2+b2 libboost-mpi-python1.83-dev_1.83.0-4.2 libboost-mpi-python1.83.0_1.83.0-4.2 libboost-mpi1.83-dev_1.83.0-4.2 libboost-mpi1.83.0_1.83.0-4.2 libboost-nowide-dev_1.83.0.2+b2 libboost-nowide1.83-dev_1.83.0-4.2 libboost-nowide1.83.0_1.83.0-4.2 libboost-numpy-dev_1.83.0.2+b2 libboost-numpy1.83-dev_1.83.0-4.2 libboost-numpy1.83.0_1.83.0-4.2 libboost-program-options-dev_1.83.0.2+b2 libboost-program-options1.83-dev_1.83.0-4.2 libboost-program-options1.83.0_1.83.0-4.2 libboost-python-dev_1.83.0.2+b2 libboost-python1.83-dev_1.83.0-4.2 libboost-python1.83.0_1.83.0-4.2 libboost-random-dev_1.83.0.2+b2 libboost-random1.83-dev_1.83.0-4.2 libboost-random1.83.0_1.83.0-4.2 libboost-regex-dev_1.83.0.2+b2 libboost-regex1.83-dev_1.83.0-4.2 libboost-regex1.83.0_1.83.0-4.2 libboost-serialization-dev_1.83.0.2+b2 libboost-serialization1.83-dev_1.83.0-4.2 libboost-serialization1.83.0_1.83.0-4.2 libboost-stacktrace-dev_1.83.0.2+b2 libboost-stacktrace1.83-dev_1.83.0-4.2 libboost-stacktrace1.83.0_1.83.0-4.2 libboost-system-dev_1.83.0.2+b2 libboost-system1.83-dev_1.83.0-4.2 libboost-system1.83.0_1.83.0-4.2 libboost-test-dev_1.83.0.2+b2 libboost-test1.83-dev_1.83.0-4.2 libboost-test1.83.0_1.83.0-4.2 libboost-thread-dev_1.83.0.2+b2 libboost-thread1.83-dev_1.83.0-4.2 libboost-thread1.83.0_1.83.0-4.2 libboost-timer-dev_1.83.0.2+b2 libboost-timer1.83-dev_1.83.0-4.2 libboost-timer1.83.0_1.83.0-4.2 libboost-tools-dev_1.83.0.2+b2 libboost-type-erasure-dev_1.83.0.2+b2 libboost-type-erasure1.83-dev_1.83.0-4.2 libboost-type-erasure1.83.0_1.83.0-4.2 libboost-url-dev_1.83.0.2+b2 libboost-url1.83-dev_1.83.0-4.2 libboost-url1.83.0_1.83.0-4.2 libboost-wave-dev_1.83.0.2+b2 libboost-wave1.83-dev_1.83.0-4.2 libboost-wave1.83.0_1.83.0-4.2 libboost1.83-dev_1.83.0-4.2 libboost1.83-tools-dev_1.83.0-4.2 libbpf-dev_1:1.5.0-3 libbpf1_1:1.5.0-3 libbrotli1_1.1.0-2+b7 libbsd-dev_0.12.2-2 libbsd0_0.12.2-2 libbz2-1.0_1.0.8-6 libc-bin_2.41-9 libc-dev-bin_2.41-9 libc6_2.41-9 libc6-dev_2.41-9 libcap-dev_1:2.75-9 libcap-ng0_0.8.5-4+b1 libcap2_1:2.75-9 libcc1-0_14.2.0-19 libcdb-dev_0.81-2 libcdb1_0.81-2 libcom-err2_1.47.2-3+b1 libcrypt-dev_1:4.4.38-1 libcrypt1_1:4.4.38-1 libctf-nobfd0_2.44-3 libctf0_2.44-3 libcurl4t64_8.14.1-2 libdb5.3t64_5.3.28+dfsg2-9 libdebconfclient0_0.279 libdebhelper-perl_13.24.2 libdpkg-perl_1.22.21 libedit-dev_3.1-20250104-1 libedit2_3.1-20250104-1 libelf-dev_0.192-4 libelf1t64_0.192-4 libexpat1_2.7.1-1 libexpat1-dev_2.7.1-1 libfakeroot_1.37.1.2-1 libffi8_3.4.8-2 libfile-stripnondeterminism-perl_1.14.1-2 libfstrm-dev_0.6.1-1+b3 libfstrm0_0.6.1-1+b3 libgcc-14-dev_14.2.0-19 libgcc-s1_14.2.0-19 libgdbm-compat4t64_1.24-2 libgdbm6t64_1.24-2 libgfortran-14-dev_14.2.0-19 libgfortran5_14.2.0-19 libgit2-1.9_1.9.0+ds-2 libgmp10_2:6.3.0+dfsg-3 libgnutls30t64_3.8.9-3 libgomp1_14.2.0-19 libgssapi-krb5-2_1.21.3-5 libhogweed6t64_3.10.1-1 libhttp-parser2.9_2.9.4-6+b2 libhwloc-dev_2.12.0-4 libhwloc15_2.12.0-4 libicu-dev_76.1-4 libicu76_76.1-4 libidn2-0_2.3.8-2 libisl23_0.27-1 libjansson4_2.14-2+b3 libjs-jquery_3.6.1+dfsg+~3.5.14-1 libjs-sphinxdoc_8.1.3-5 libjs-underscore_1.13.4~dfsg+~1.11.4-3 libk5crypto3_1.21.3-5 libkeyutils1_1.6.3-6 libkrb5-3_1.21.3-5 libkrb5support0_1.21.3-5 liblastlog2-2_2.41-5 libldap2_2.6.10+dfsg-1 libllvm19_1:19.1.7-3+b1 liblmdb-dev_0.9.31-1+b2 liblmdb0_0.9.31-1+b2 libltdl-dev_2.5.4-4 libltdl7_2.5.4-4 liblua5.3-0_5.3.6-2+b4 liblua5.3-dev_5.3.6-2+b4 liblua5.4-0_5.4.7-1+b2 liblz4-1_1.10.0-4 liblzma5_5.8.1-1 libmagic-mgc_1:5.46-5 libmagic1t64_1:5.46-5 libmariadb3_1:11.8.2-1 libmbedcrypto16_3.6.3-1 libmbedtls21_3.6.3-1 libmbedx509-7_3.6.3-1 libmd-dev_1.1.0-2+b1 libmd0_1.1.0-2+b1 libmount1_2.41-5 libmpc3_1.3.1-1+b3 libmpfr6_4.2.2-1 libmpich-dev_4.3.0+really4.2.1-1 libmpich12_4.3.0+really4.2.1-1 libncurses-dev_6.5+20250216-2 libncurses6_6.5+20250216-2 libncursesw6_6.5+20250216-2 libnetsnmptrapd40t64_5.9.4+dfsg-2 libnettle8t64_3.10.1-1 libnghttp2-14_1.64.0-1.1 libnghttp2-dev_1.64.0-1.1 libnghttp3-9_1.8.0-1 libnuma-dev_2.0.19-1 libnuma1_2.0.19-1 libp11-kit0_0.25.5-3 libpam-modules_1.7.0-5 libpam-modules-bin_1.7.0-5 libpam-runtime_1.7.0-5 libpam0g_1.7.0-5 libpci-dev_1:3.13.0-2 libpci3_1:3.13.0-2 libpcre2-8-0_10.45-1 libperl5.40_5.40.1-5 libpipeline1_1.5.8-1 libpkgconf3_1.8.1-4 libproc2-0_2:4.0.4-8 libprotobuf-dev_3.21.12-11 libprotobuf-lite32t64_3.21.12-11 libprotobuf32t64_3.21.12-11 libprotoc32t64_3.21.12-11 libpsl5t64_0.21.2-1.1+b1 libpython3-dev_3.13.5-1 libpython3-stdlib_3.13.5-1 libpython3.13_3.13.5-2 libpython3.13-dev_3.13.5-2 libpython3.13-minimal_3.13.5-2 libpython3.13-stdlib_3.13.5-2 libre-engine-re2-perl_0.18+ds-2 libre2-11_20240702-3+b1 libre2-dev_20240702-3+b1 libreadline-dev_8.2-6 libreadline8t64_8.2-6 librtmp1_2.4+20151223.gitfa8646d.1-2+b5 librust-adler-dev_1.0.2-2+b2 librust-ahash-dev_0.8.11-9 librust-allocator-api2-dev_0.2.16-2 librust-arbitrary-dev_1.4.1-1 librust-byteorder-dev_1.5.0-1+b2 librust-cc-dev_1.2.19-1 librust-cfg-if-dev_1.0.0-1+b2 librust-codespan-reporting-dev_0.11.1-2+b2 librust-compiler-builtins-dev_0.1.139-1 librust-const-random-dev_0.1.17-2 librust-const-random-macro-dev_0.1.16-2 librust-crc32fast-dev_1.4.2-1 librust-critical-section-dev_1.2.0-1 librust-crossbeam-deque-dev_0.8.5-1 librust-crossbeam-epoch+std-dev_0.9.18-1 librust-crossbeam-epoch-dev_0.9.18-1 librust-crossbeam-utils-dev_0.8.19-1 librust-crunchy-dev_0.2.2-1+b2 librust-cxx-build-dev_1.0.141-1 librust-cxx-dev_1.0.141-1 librust-cxxbridge-flags-dev_1.0.141-1 librust-cxxbridge-macro-dev_1.0.141-1 librust-derive-arbitrary-dev_1.4.1-1+b1 librust-either-dev_1.13.0-1 librust-equivalent-dev_1.0.1-1+b2 librust-erased-serde-dev_0.3.31-1 librust-flate2-dev_1.1.0-1 librust-foldhash-dev_0.1.4-1 librust-getrandom-dev_0.2.15-1 librust-hashbrown-dev_0.14.5-5 librust-indexmap-dev_2.7.0-1 librust-itoa-dev_1.0.14-1 librust-jobserver-dev_0.1.32-1 librust-libc-dev_0.2.169-1 librust-libz-sys-dev_1.1.20-1+b2 librust-link-cplusplus-dev_1.0.10-1 librust-log-dev_0.4.26-1 librust-memchr-dev_2.7.4-1 librust-memmap-dev_0.7.0-1+b2 librust-miniz-oxide-dev_0.7.1-2 librust-no-panic-dev_0.1.32-1 librust-once-cell-dev_1.20.2-1 librust-parking-lot-core-dev_0.9.10-1 librust-pkg-config-dev_0.3.31-1 librust-portable-atomic-dev_1.10.0-3 librust-proc-macro2-dev_1.0.92-1+b1 librust-quote-dev_1.0.40-1 librust-rayon-core-dev_1.12.1-1 librust-rayon-dev_1.10.0-1 librust-rustc-std-workspace-core-dev_1.0.0-1+b2 librust-rustversion-dev_1.0.14-1+b2 librust-ryu-dev_1.0.19-1 librust-scratch-dev_1.0.7-1 librust-serde-derive-dev_1.0.217-1 librust-serde-dev_1.0.217-1 librust-serde-fmt-dev_1.0.3-4 librust-serde-json-dev_1.0.139-1 librust-serde-test-dev_1.0.171-1+b2 librust-serde-yaml-dev_0.9.34-1 librust-shlex-dev_1.3.0-1 librust-simd-adler32-dev_0.3.7-1+b2 librust-smallvec-dev_1.13.2-1 librust-sval-buffer-dev_2.6.1-1+b3 librust-sval-derive-dev_2.6.1-2+b2 librust-sval-dev_2.6.1-2+b2 librust-sval-dynamic-dev_2.6.1-1+b3 librust-sval-fmt-dev_2.6.1-1+b2 librust-sval-ref-dev_2.6.1-1+b3 librust-sval-serde-dev_2.6.1-1+b3 librust-syn-dev_2.0.98-1 librust-termcolor-dev_1.4.1-1 librust-tiny-keccak-dev_2.0.2-1+b3 librust-unicode-ident-dev_1.0.13-1 librust-unicode-width-0.1-dev_0.1.14-2+b2 librust-unsafe-libyaml-dev_0.2.11-1 librust-value-bag-dev_1.9.0-1 librust-value-bag-serde1-dev_1.9.0-1 librust-value-bag-sval2-dev_1.9.0-1 librust-version-check-dev_0.9.5-1 librust-winapi-dev_0.3.9-1+b3 librust-winapi-i686-pc-windows-gnu-dev_0.4.0-1+b2 librust-winapi-util-dev_0.1.6-1+b2 librust-winapi-x86-64-pc-windows-gnu-dev_0.4.0-1+b2 librust-zerocopy-derive-dev_0.7.34-1 librust-zerocopy-dev_0.7.34-1 libsasl2-2_2.1.28+dfsg1-9 libsasl2-modules-db_2.1.28+dfsg1-9 libseccomp2_2.6.0-2 libselinux1_3.8.1-1 libsemanage-common_3.8.1-1 libsemanage2_3.8.1-1 libsensors-config_1:3.6.2-2 libsensors-dev_1:3.6.2-2 libsensors5_1:3.6.2-2 libsepol2_3.8.1-1 libsframe1_2.44-3 libslurm42t64_24.11.5-3 libsmartcols1_2.41-5 libsnmp-base_5.9.4+dfsg-2 libsnmp-dev_5.9.4+dfsg-2 libsnmp40t64_5.9.4+dfsg-2 libsodium-dev_1.0.18-1+b2 libsodium23_1.0.18-1+b2 libsqlite3-0_3.46.1-6 libssh2-1t64_1.11.1-1 libssl-dev_3.5.0-2 libssl3t64_3.5.0-2 libstd-rust-1.85_1.85.0+dfsg3-1 libstd-rust-dev_1.85.0+dfsg3-1 libstdc++-14-dev_14.2.0-19 libstdc++6_14.2.0-19 libsystemd-dev_257.7-1 libsystemd-shared_257.7-1 libsystemd0_257.7-1 libtasn1-6_4.20.0-2 libtinfo6_6.5+20250216-2 libtool_2.5.4-4 libubsan1_14.2.0-19 libuchardet0_0.0.8-1+b2 libudev-dev_257.7-1 libudev1_257.7-1 libunistring5_1.3-2 libuuid1_2.41-5 libwrap0_7.6.q-36 libwrap0-dev_7.6.q-36 libxdp-dev_1.5.4-1 libxdp1_1.5.4-1 libxml2_2.12.7+dfsg+really2.9.14-1 libxxhash0_0.8.3-2 libyaml-0-2_0.2.5-2 libz3-4_4.13.3-1 libzstd-dev_1.5.7+dfsg-1 libzstd1_1.5.7+dfsg-1 linux-libc-dev_6.12.35-1 login_1:4.16.0-2+really2.41-5 login.defs_1:4.17.4-2 m4_1.4.19-8 make_4.4.1-2 man-db_2.13.1-1 mariadb-common_1:11.8.2-1 mawk_1.3.4.20250131-1 media-types_13.0.0 mount_2.41-5 mpi-default-bin_1.18 mpi-default-dev_1.18 mpich_4.3.0+really4.2.1-1 mysql-common_5.8+1.1.1 ncurses-base_6.5+20250216-2 ncurses-bin_6.5+20250216-2 netbase_6.5 openssl_3.5.0-2 openssl-provider-legacy_3.5.0-2 passwd_1:4.17.4-2 patch_2.8-2 pci.ids_0.0~2025.06.09-1 perl_5.40.1-5 perl-base_5.40.1-5 perl-modules-5.40_5.40.1-5 pkgconf_1.8.1-4 pkgconf-bin_1.8.1-4 po-debconf_1.0.21+nmu1 policy-rcd-declarative_0.6 policy-rcd-declarative-deny-all_0.6 procps_2:4.0.4-8 protobuf-compiler_3.21.12-11 python3_3.13.5-1 python3-dev_3.13.5-1 python3-minimal_3.13.5-1 python3-yaml_6.0.2-1+b2 python3.13_3.13.5-2 python3.13-dev_3.13.5-2 python3.13-minimal_3.13.5-2 ragel_6.10-5 readline-common_8.2-6 rpcsvc-proto_1.4.3-1 rustc_1.85.0+dfsg3-1 sbuild-build-depends-main-dummy_0.invalid.0 sed_4.9-2 sensible-utils_0.0.25 sqv_1.3.0-3 systemd_257.7-1 sysvinit-utils_3.14-4 tar_1.35+dfsg-3.1 tzdata_2025b-4 util-linux_2.41-5 xz-utils_5.8.1-1 zlib1g_1:1.3.dfsg+really1.3.1-1+b1 zlib1g-dev_1:1.3.dfsg+really1.3.1-1+b1

--- a/pdns/dnsdistdist/dnsdist-lua-configuration-items.cc
+++ b/pdns/dnsdistdist/dnsdist-lua-configuration-items.cc
@@ -38,7 +38,7 @@ struct BooleanConfigurationItems
 struct UnsignedIntegerConfigurationItems
 {
   const std::function<void(dnsdist::configuration::RuntimeConfiguration& config, uint64_t value)> mutator;
-  const size_t maximumValue{std::numeric_limits<uint64_t>::max()};
+  const uint64_t maximumValue{std::numeric_limits<uint64_t>::max()};
 };
 
 struct StringConfigurationItems
@@ -53,7 +53,7 @@ struct BooleanImmutableConfigurationItems
 struct UnsignedIntegerImmutableConfigurationItems
 {
   const std::function<void(dnsdist::configuration::ImmutableConfiguration& config, uint64_t value)> mutator;
-  const size_t maximumValue{std::numeric_limits<uint64_t>::max()};
+  const uint64_t maximumValue{std::numeric_limits<uint64_t>::max()};
 };
 
 struct DoubleImmutableConfigurationItems

--- a/pdns/dnsdistdist/dnsdist-lua.cc
+++ b/pdns/dnsdistdist/dnsdist-lua.cc
@@ -253,7 +253,7 @@ static void parseTLSConfig(TLSConfig& config, const std::string& context, boost:
 
 #endif // defined(HAVE_DNS_OVER_TLS) || defined(HAVE_DNS_OVER_HTTPS)
 
-void checkParameterBound(const std::string& parameter, uint64_t value, size_t max)
+void checkParameterBound(const std::string& parameter, uint64_t value, uint64_t max)
 {
   if (value > max) {
     throw std::runtime_error("The value (" + std::to_string(value) + ") passed to " + parameter + " is too large, the maximum is " + std::to_string(max));

--- a/pdns/dnsdistdist/dnsdist-lua.hh
+++ b/pdns/dnsdistdist/dnsdist-lua.hh
@@ -42,7 +42,7 @@ using luadnsrule_t = boost::variant<string, LuaArray<std::string>, std::shared_p
 std::shared_ptr<DNSRule> makeRule(const luadnsrule_t& var, const std::string& calledFrom);
 
 void parseRuleParams(boost::optional<luaruleparams_t>& params, boost::uuids::uuid& uuid, std::string& name, uint64_t& creationOrder);
-void checkParameterBound(const std::string& parameter, uint64_t value, size_t max = std::numeric_limits<uint16_t>::max());
+void checkParameterBound(const std::string& parameter, uint64_t value, uint64_t max = std::numeric_limits<uint16_t>::max());
 
 void setupLua(LuaContext& luaCtx, bool client, bool configCheck, const std::string& config);
 void setupLuaActions(LuaContext& luaCtx);

--- a/pdns/dnsdistdist/test-dnsdistrules_cc.cc
+++ b/pdns/dnsdistdist/test-dnsdistrules_cc.cc
@@ -12,7 +12,7 @@
 #include "dnsdist-rules.hh"
 #include "dnsdist-rules-factory.hh"
 
-void checkParameterBound(const std::string& parameter, uint64_t value, size_t max)
+void checkParameterBound(const std::string& parameter, uint64_t value, uint64_t max)
 {
   if (value > max) {
     throw std::runtime_error("The value passed to " + parameter + " is too large, the maximum is " + std::to_string(max));

